src/osd/PGLog.h

```diff
@@ -88,8 +88,13 @@ struct PGLog : DoutPrefixProvider {
     mempool::osd_pglog::list<pg_log_entry_t>::reverse_iterator
       rollback_info_trimmed_to_riter;
 
+    /*
+     * return true if we need to mark the pglog as dirty
+     */
     template <typename F>
-    void advance_can_rollback_to(eversion_t to, F &&f) {
+    bool advance_can_rollback_to(eversion_t to, F &&f) {
+      bool dirty = to > can_rollback_to || to > rollback_info_trimmed_to;
```

Comment:
@Aran85 rollback_info_trimmed_to <= can_rollback_to, so you only need to check against rollback_info_trimmed_to

Reply:
so we need to check against can_rollback_to ?